pkg/diagnostics/collectors.go

```diff
@@ -65,6 +65,8 @@ func (c *collectorFactory) DataCenterConfigCollectors(datacenter v1alpha1.Ref) [
 		return c.eksaVsphereCollectors()
 	case v1alpha1.DockerDatacenterKind:
 		return c.eksaDockerCollectors()
+	case v1alpha1.CloudStackDatacenterKind:
```

Comment:
I wasn't able to define unit tests here without making a lot of these types public to the test package